src/main/scala/org/broadinstitute/dsde/agora/server/dataaccess/permissions/PermissionsClient.scala

```diff
@@ -1,98 +1,86 @@
 package org.broadinstitute.dsde.agora.server.dataaccess.permissions
 
 import AgoraPermissions._
-import org.broadinstitute.dsde.agora.server.AgoraConfig
+import org.broadinstitute.dsde.agora.server.dataaccess.{ReadAction, ReadWriteAction, WriteAction}
 import org.broadinstitute.dsde.agora.server.exceptions.PermissionNotFoundException
 import org.broadinstitute.dsde.agora.server.model.AgoraEntity
+import slick.jdbc.JdbcProfile
 
-import scala.concurrent.{Await, Future}
 import scala.concurrent.ExecutionContext.Implicits.global
 import scala.concurrent.duration._
-import scala.util.Try
+import scala.util.{Failure, Success, Try}
 
-trait PermissionsClient {
-  import AgoraConfig.sqlDatabase.profile.api._
-  val db = AgoraConfig.sqlDatabase.db
+abstract class PermissionsClient(profile: JdbcProfile) {
+  import profile.api._
 
   val timeout = 10.seconds
 
   def alias(entity: AgoraEntity): String
 
+  def withPermissionNotFoundException[T](errorString: String = "Could not get permission")( op: => ReadWriteAction[T] ): ReadWriteAction[T] = {
+    op.asTry flatMap {
+      case Success(permission) => DBIO.successful(permission)
+      case Failure(ex) => DBIO.failed(new PermissionNotFoundException(errorString, ex))
+    }
+  }
+
   // Users
-  def addUserIfNotInDatabase(userEmail: String): Unit = {
+  def addUserIfNotInDatabase(userEmail: String): WriteAction[Int] = {
     // Attempts to add user to UserTable and ignores errors if user already exists
-    try {
-      Await.ready(db.run(users += UserDao(userEmail)), timeout)
-    } catch {
-      case _ : Throwable => //Do nothing
+    (users += UserDao(userEmail)).asTry flatMap {
+      case Success(count) => DBIO.successful(count)
+      case Failure(_) => DBIO.successful(0)
```

Comment:
This does what it used to do (and more... nice use of `0`). I worry that failure could mean something other than the user already exists.

I suspect there may be many things like this in this review. I don't think it makes sense to fix it in the context of this sweeping code improvement. How do you want to handle issues like these?